src/WebJobs.Script.WebHost/Metrics/FlexConsumptionMetricsPublisher.cs

```diff
@@ -260,11 +271,28 @@ public void OnFunctionCompleted(string functionName, string invocationId)
 
```

Comment:
When would this ever be less than 0 when this code is invoked? Are we expecting we may run into scenarios where `OnFunctionCompleted` was called without `OnFunctionStarted`? If so, should we emit a log here?

Reply:
You're right that it should never be called when active count == 0. It would mean the upstream MetricsManager that is calling this interface method is out of sync. I notice that even in that class, we're just decrementing _activeFunctionCount blindly and assuming things are being managed correctly further upstream.

I'm not going to make any changes here to guard against this in this PR. Anything we do should be done consistently for all metrics publishers, and upstream as well.